### DISCUSSION TITLE
Column resize problem

### DISCUSSION THREAD

**Discussion by gmmoliveros** [TanStack/table#5396] [Q&A] [answered]
Is it possible to implement column resize without using inline styles in react table v8 ? I imitated how the performant sample works inthr docs but on my code review we want to avoid inline styles. Is there any way to do that? We also don't use css in js. Thanks!

**Comment by KevinVandy** [comment]
Yeah, for sure. I don't think most people use inline styles at all. We set up the examples that way because it's just a neutral way to show the required styles all in 1 file in a code sandbox.

Though, for the column widths themselves from column.getSize(), it doesn't make much of a difference if using a direct style or a CSS variable. I've always said that a couple inline styles here and there for frequently changing values are fine, if not better in some situations.

**Reply by gmmoliveros** [reply]
How can I have a column resizing table without using inline styles? Sorry just asking for ideas because i can't wrap my head around having javascript computed values applied to styles without having inline styles or something like styled components. Thank you!

**Reply by gmmoliveros** [reply]
Thanks for your answer but can you point me in the right path? Maybe in the docs on how will i be able to implement resizing without using inline styles? Thank you so much.

**Reply by KevinVandy** [reply]
CSS vars

**Reply by KevinVandy** [reply]
The performant column resizing example uses CSS vars

**Reply by gmmoliveros** [reply]
That would require to have a single inline style when placing/declaring all css vars in the table element. Can we avoid that or not? Thank you so much!

**Reply by KevinVandy** [reply, marked as answer]
Ok. Please explain why you need to avoid  inline styles at all cost. Did you hear from someone that they are bad practice or something? They are only bad practice if they are overused. They are useful and necessary for lots of use cases, including this.

**Reply by gmmoliveros** [reply]
Thank you! I understand and will continue implementing it using CSS vars. Thank you for the library! 👍👍
